lib/ansible/modules/windows/win_audit_policy_system.ps1

```diff
@@ -0,0 +1,129 @@
+#!powershell
+# Copyright (c) 2017 Noah Sparks <nsparks@outlook.com>
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy.psm1
+#Requires -Module Ansible.ModuleUtils.CommandUtil.psm1
+
+$ErrorActionPreference = 'Stop'
+
+$params = Parse-Args -arguments $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+
+$results = @{
+    changed = $false
+}
+
+######################################
+### populate sets for -validateset ###
+######################################
+$categories_rc = run-command -command 'auditpol /list /category /r'
+$subcategories_rc = run-command -command 'auditpol /list /subcategory:* /r'
+
+If ($categories_rc.item('rc') -eq 0)
+{
+    $categories = ConvertFrom-Csv $categories_rc.item('stdout') | Select-Object -expand Category*
+}
+Else
+{
+    Fail-Json -obj $results -message "Failed to retrive audit policy categories. Please make sure the auditpol command is functional on
+    the system and that the account ansible is running under is able to retrieve them. $($_.Exception.Message)"
+}
+
+If ($subcategories_rc.item('rc') -eq 0)
+{
+    $subcategories = ConvertFrom-Csv $subcategories_rc.item('stdout') | Select-Object -expand Category* |
+    Where-Object {$_ -notin $categories}
+}
+Else
+{
+    Fail-Json -obj $results -message "Failed to retrive audit policy subcategories. Please make sure the auditpol command is functional on
+    the system and that the account ansible is running under is able to retrieve them. $($_.Exception.Message)"
+}
+
+######################
+### ansible params ###
+######################
+$category = Get-AnsibleParam -obj $params -name "category" -type "str" -ValidateSet $categories
+$subcategory = Get-AnsibleParam -obj $params -name "subcategory" -type "str" -ValidateSet $subcategories
+$audit_type = Get-AnsibleParam -obj $params -name "audit_type" -type "str" -ValidateSet 'success', 'failure', 'success and failure', 'none'
+
+########################
+### Start Processing ###
+########################
+Function Get-AuditPolicy ($GetString) {
+    $auditpolcsv = Run-Command -command $GetString
+    If ($auditpolcsv.item('rc') -eq 0)
+    {
+        $Obj = ConvertFrom-CSV $auditpolcsv.item('stdout') | Select-Object @{n='subcategory';e={$_.Subcategory.ToLower()}},
+        @{n='audit_type';e={$_."Inclusion Setting".ToLower()}}
+    }
+    Else {
+        return $auditpolcsv.item('stderr')
+    }
+
+    $HT = @{}
+    Foreach ( $Item in $Obj )
+    {
+        $HT.Add($Item.subcategory,$Item.audit_type)
+    }
+    $HT
+}
+
+################################
+### Deal with parameter sets ###
```

Comment:
I personally don't like that style for comments. And it may trigger issues when we start doing style validation in the future ;-)

Reply:
thanks for checking it. You'd prefer just simple like this? 

`# comment here`

I like the fatter blocks to split sections up, but I don't mind changing it.

Reply:
I'll leave it up to others to decided, as it is a personal opinion. (That's why I didn't ask for a change in this case).

Reply:
Ok gotcha